examples/embed-directive/embed-directive.go

```diff
@@ -0,0 +1,51 @@
+// Starting in Go version 1.16+, Go supports use of `//go:embed`
+// which is a directive to embed files and folders into
+// the application binary by the compiler.
+
+// Since Go does not have preprocessor or macros like C type of languages,
+// `pragmas/directives` are implemented by the compiler as comments `//`.
+// Using `//go:embed` tells the compiler to include arbitrary files/folders in binary.
+package main
+
+// If you are not using any exported identifiers from `embed` package directly,
+// you can add an underscore `_` before the package name, ask Go to import `embed`
+// without directly using it, like this: `_ "embed"`. In this example, we remove the
+// underscore as we need to use the `embed.FS` type from the package.
+import (
+	"embed"
+)
+
+// Since one file is defined after the directive, the compiler will only include
+// this single file, followed by variable `single_file_string` to access as `string` type.
+//go:embed embed-directive.sh
+var single_file_string string
+
+// Here is a similar example but include single file as `[]byte`.
+// Note that `//go:embed` can only be used in package level,
+// define it in any function will result in error.
+//go:embed embed-directive.sh
+var single_file_byte []byte
+
+// We can also embed multiple files or even folders with wildcard.
```

Comment:
It's a clever idea to embed the .sh and .hash files, but it could be confusing because the readers of gobyexample are not aware of these files. Also, will this actually work on the playground? Each example has a generated playground link -- did you try to run the one generated for this new example?

I'm not sure how to overcome this...

Reply:
It will not be able to run on the playground just like the reading file example on your website.

Reply:
Unless go playground gives us any existing files that we can access this will not run on go playground, as embed requires pre-existing files to be presented before you can compile.

Reply:
I can however in the `.sh` file do some creation like
```
echo "hello go" > example_single_file.txt
mkdir -p example_folder
echo "123" > example_folder/example1.hash
echo "456" > example_folder/example1.hash

go run embed-directive.go
```

Thoughts?

Reply:
Ah good point about the reading files example.

Yes, having it laid out in the `.sh` makes more sense to me - good idea! I mostly feel uncomfortable exposing the "implementation detail" of these `.hash` files to users.

Reply:
Will make some changes on that.
Thanks.